packages/remix-dev/compiler/js/plugins/browserNodeBuiltinsPolyfill.ts

```diff
@@ -0,0 +1,37 @@
+import { nodeModulesPolyfillPlugin } from "esbuild-plugins-node-modules-polyfill";
+
+import type { Context } from "../../context";
+
+export const browserNodeBuiltinsPolyfillPlugin = (ctx: Context) =>
+  nodeModulesPolyfillPlugin({
+    // Rename plugin to improve error message attribution
+    name: "browser-node-builtins-polyfill-plugin",
+    // Only pass through the "modules" and "globals" options to ensure we
+    // don't leak the full plugin API to Remix consumers.
+    modules: ctx.config.browserNodeBuiltinsPolyfill?.modules ?? {},
+    globals: ctx.config.browserNodeBuiltinsPolyfill?.globals ?? {},
+    // Mark any unpolyfilled Node builtins in the build output as errors.
+    fallback: "error",
+    formatError({ moduleName, importer, polyfillExists }) {
```

Comment:
I added this functionality here: https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill/pull/153.